src/ReactWrapperComponent.jsx

```diff
@@ -8,45 +8,75 @@ import React, { PropTypes } from 'react';
  * the DOM node it rendered to, so we can't really "re-render" to
  * pass new props in.
  */
-export default class ReactWrapperComponent extends React.Component {
+export default function createWrapperComponent(node, options = {}) {
+  const spec = {
 
-  constructor(props) {
-    super(props);
-    this.state = Object.assign({}, props.props);
-  }
+    propTypes: {
+      Component: PropTypes.oneOfType([PropTypes.func, PropTypes.string]).isRequired,
+      props: PropTypes.object.isRequired,
+      context: PropTypes.object,
+    },
 
-  setProps(newProps) {
-    return new Promise(resolve => this.setState(newProps, resolve));
-  }
+    getDefaultProps() {
+      return {
+        context: null,
+      };
+    },
+
+    getInitialState() {
+      return {
+        props: this.props.props,
+        context: this.props.context,
+      };
+    },
+
+    setChildProps(props) {
+      return new Promise(resolve => this.setState({ props }, resolve));
```

Comment:
because `this.setState()` can or maybe will be an asynchronous call... but it doesn't seem to be in react 0.13 or 0.14...  originally I had set up the API in such a way that it returned a promise, but it doesn't appear to be needed in most cases (and is thus not used right now). This promise code could be removed (maybe separate PR)